client/my-sites/email/titan-mail-add-mailboxes/index.jsx

```diff
@@ -0,0 +1,305 @@
+						return;
+					}
+					return this.isMounted && page( '/checkout/' + selectedSite.slug );
+				} );
+		}
+	};
+
+	handleUnusedMailboxFinishSetupClick = () => {
+		const {
+			currentRoute,
+			isSelectedDomainNameValid,
+			selectedDomainName,
+			selectedSite,
+		} = this.props;
+
+		this.recordClickEvent( 'calypso_email_management_titan_add_mailboxes_create_mailbox_click' );
+
+		const domainName = isSelectedDomainNameValid ? selectedDomainName : null;
+
+		if ( isEnabled( 'titan/iframe-control-panel' ) ) {
+			page(
+				emailManagementManageTitanAccount( selectedSite.slug, domainName, currentRoute, {
+					context: TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL,
+				} )
+			);
+
+			return;
+		}
+
+		window.open(
+			emailManagementTitanControlPanelRedirect( selectedSite.slug, domainName, currentRoute, {
+				context: TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL,
+			} )
+		);
+	};
+
+	onButtonClick = () => {
+		this.setState( { quantity: this.state.quantity + 1 } );
+	};
+
+	onMailboxesChange = ( updatedMailboxes ) => {
+		this.setState( { mailboxes: updatedMailboxes, quantity: updatedMailboxes.length } );
+	};
+
+	renderForm() {
+		const { isLoadingDomains, selectedDomainName, titanMonthlyProduct, translate } = this.props;
+
+		if ( isLoadingDomains || ! titanMonthlyProduct ) {
+			return <AddEmailAddressesCardPlaceholder />;
+		}
+
+		return (
+			<>
+				<SectionHeader label={ translate( 'Add New Mailboxes' ) } />
+
+				<Card>
+					<TitanNewMailboxList
+						domain={ selectedDomainName }
+						mailboxes={ this.state.mailboxes }
+						onMailboxesChange={ this.onMailboxesChange }
+					>
+						<Button
+							className="titan-mail-add-mailboxes__action-cancel"
+							onClick={ this.handleCancel }
+						>
+							{ translate( 'Cancel' ) }
+						</Button>
+						<Button
+							className="titan-mail-add-mailboxes__action-continue"
+							primary
+							onClick={ this.handleContinue }
+						>
+							{ translate( 'Continue' ) }
+						</Button>
+					</TitanNewMailboxList>
+				</Card>
+			</>
+		);
+	}
+
+	render() {
+		const {
+			domainsWithForwards,
+			selectedDomain,
+			selectedDomainName,
+			selectedSite,
+			isSelectedDomainNameValid,
+			isLoadingDomains,
+			maxTitanMailboxCount,
+			titanMonthlyProduct,
+		} = this.props;
+
+		if ( ! isLoadingDomains && ! isSelectedDomainNameValid ) {
+			this.goToEmail();
+			return null;
+		}
+
+		const finishSetupLinkIsExternal = ! isEnabled( 'titan/iframe-control-panel' );
+
+		return (
+			<>
+				<QueryProductsList />
+
+				{ selectedSite && <QuerySiteDomains siteId={ selectedSite.ID } /> }
+
+				<Main>
+					<DomainManagementHeader
+						onClick={ this.goToEmail }
+						selectedDomainName={ selectedDomainName }
+					>
+						{ getTitanProductName() + ': ' + selectedDomainName }
+					</DomainManagementHeader>
+
+					<TitanExistingForwardsNotice domainsWithForwards={ domainsWithForwards } />
+					<TitanUnusedMailboxesNotice
+						domain={ selectedDomain }
```

Comment:
FYI, this property (that is marked as required) threw a warning when loading the component with an empty redux tree:

```
Warning: Failed prop type: The prop `domain` is marked as required in `TitanUnusedMailboxesNotice`, but its value is `undefined`.
    in TitanUnusedMailboxesNotice (created by TitanMailAddMailboxes)
    in TitanMailAddMailboxes (created by Localized(TitanMailAddMailboxes))
    in Localized(TitanMailAddMailboxes) (created by Context.Consumer)
    in WithLocalizedMoment(Localized(TitanMailAddMailboxes)) (created by ShoppingCartWrapper)
    in ShoppingCartWrapper (created by ConnectFunction)
    in ConnectFunction
...
```

Not a blocker, just good to avoid for the future.

Reply:
I've made this conditional in the caller.